edgeless_api/proto/messages.proto

```diff
@@ -98,8 +98,24 @@ message UpdateNodeRequest {
     optional string agent_url                                 = 3;
     // URL of the node's invocation server. Always present if REGISTER type.
     optional string invocation_url                            = 4;
-    // Resources offered by this node.
+    // Resources offered by this node. May be only present if REGISTER type.
     repeated ResourceProviderSpecification resource_providers = 5;
+    // Node capabilities. May be only present if REGISTER type.
+    NodeCapabilities capabilities                             = 6;
+}
+
+// Node capabilities exposed to the orchestrator.
+message NodeCapabilities {
+    // Number of (actual or virtual) CPUs associated with the edge node.
+    uint32 num_cpus       = 10;
```

Comment:
Why are we starting with 10 here?
Do you have a plan for 1--9?

Reply:
That is for future growth: when a new field is added, it can be added "closers" to (e.g.) CPU stuff by assigning a number in 10-19.

If you don't like this, I will be very happy to use incremental numbers, please let me know your opinion

Reply:
This is fine. I just didn't know what you had in mind there.
Might be worth documenting somewhere.